### PR TITLE
chore(H2): onboard H2 into Undici queueing system

### PR DESCRIPTION
**Pull request by metcoder95**
## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: test/http2.js [hunk 1]
```diff
@@ -1325,62 +1269,82 @@ test('#2364 - Concurrent aborts', async t => {
...
+    (err, response) => {
+      console.log('with response')
+      t.ifError(err)
+      t.strictEqual(
+        response.headers['content-type'],
+        'text/plain; charset=utf-8'
+      )
+      t.strictEqual(response.headers['x-custom-h2'], 'hello')
+      t.strictEqual(response.statusCode, 200)
+      // response.body.dump()
```
Comment (metcoder95) [inline test/http2.js:1330]:
```suggestion
```

File: test/http2.js [hunk 2]
```diff
@@ -1325,62 +1269,82 @@ test('#2364 - Concurrent aborts', async t => {
...
```
Comment (metcoder95) [inline test/http2.js:1294]:
```suggestion
```

File: test/http2.js [hunk 3]
```diff
@@ -1439,10 +1403,56 @@ test('#3046 - GOAWAY Frame', async t => {
...
+  t.rejects(response.body.text(), {
     message: 'HTTP/2: "GOAWAY" frame received with code 0',
-    code: 'UND_ERR_ABORTED'
+    code: 'UND_ERR_SOCKET'
+  })
+
+  await t.completed
+})
+
+test('#3671 - Graceful close', { only: true }, async (t) => {
```
Comment (metcoder95) [inline test/http2.js:1414]:
```suggestion
test('#3671 - Graceful close', async (t) => {
```

File: test/http2.js [hunk 4]
```diff
@@ -1325,62 +1269,80 @@ test('#2364 - Concurrent aborts', async t => {
...
-    t.strictEqual(err.name, 'AbortError')
-  })
-
-  client.request({
-    path: '/3',
-    method: 'GET',
-    headers: {
-      'x-my-header': 'foo'
+    (err, response) => {
+      console.log('without response')
```
Comment (metcoder95) [inline test/http2.js:1307]:
```suggestion
```

File: test/http2.js [hunk 5]
```diff
@@ -1325,62 +1269,80 @@ test('#2364 - Concurrent aborts', async t => {
...
+  client.request(
+    {
+      path: '/1',
+      method: 'GET',
+      headers: {
+        'x-my-header': 'foo'
+      }
+    },
+    (err, response) => {
+      console.log('with response')
```
Comment (metcoder95) [inline test/http2.js:1286]:
```suggestion
```

File: test/http2.js [hunk 6]
```diff
@@ -1325,62 +1269,80 @@ test('#2364 - Concurrent aborts', async t => {
...
+      headers: {
+        'x-my-header': 'foo'
+      }
     },
-    signal: controller.signal
-  }, (err, response) => {
-    t.strictEqual(err.name, 'AbortError')
-  })
+    (err, response) => {
+      console.log('with response')
```
Comment (metcoder95) [inline test/http2.js:1321]:
```suggestion
```

File: test/http2.js [hunk 7]
```diff
@@ -1325,62 +1269,80 @@ test('#2364 - Concurrent aborts', async t => {
...
+    {
+      path: '/4',
+      method: 'GET',
+      headers: {
+        'x-my-header': 'foo'
+      },
+      signal
+    },
+    (err, response) => {
+      console.log('without response')
```
Comment (metcoder95) [inline test/http2.js:1342]:
```suggestion
```

**Review by ronag — APPROVED** [review]

**Comment by metcoder95** [comment]
Are we good to merge @mcollina?

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**